src/Illuminate/Validation/Concerns/ValidatesAttributes.php

```diff
@@ -407,7 +407,7 @@ public function validateBetween($attribute, $value, $parameters)
 
         $size = $this->getSize($attribute, $value);
 
-        return $size >= $parameters[0] && $size <= $parameters[1];
+        return $size >= (int) str_replace('_', '', (string) $parameters[0]) && $size <= (int) str_replace('_', '', (string) $parameters[1]);
```

Comment:
They should be casted to float, as the boundaries might not be integers.

Reply:
Is it better to remove the (int) cast and leave it to PHP?

Reply:
I would add extra tests to cover this case and adjust your code accordingly

Reply:
I force pushed some more tests to cover the floats.